### PR TITLE
TSK-448: removed spring-boot from taskana-rest-spring and taskana-rest

### PR DESCRIPTION
**Pull request by mustaphazorgati**

### CODE REVIEW

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/RestConfiguration.java [hunk 1]
```diff
@@ -37,19 +34,10 @@
...
-    @ConfigurationProperties(prefix = "datasource")
-    public DataSourceProperties dataSourceProperties() {
-        DataSourceProperties props = new DataSourceProperties();
-        props.setUrl("jdbc:h2:mem:taskana;IGNORECASE=TRUE;LOCK_MODE=0;INIT=CREATE SCHEMA IF NOT EXISTS TASKANA");
-        return props;
-    }
-
-    @Bean
-    @Primary
-    public DataSource dataSource(DataSourceProperties properties) {
```
Comment (TitusLabs) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/RestConfiguration.java]: please remove the tx manager for the datasource too

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/RestConfiguration.java [hunk 2]
```diff
@@ -37,19 +34,10 @@
 @EnableTransactionManagement
 public class RestConfiguration {
 
-    @Bean
-    @Primary
-    @ConfigurationProperties(prefix = "datasource")
-    public DataSourceProperties dataSourceProperties() {
-        DataSourceProperties props = new DataSourceProperties();
-        props.setUrl("jdbc:h2:mem:taskana;IGNORECASE=TRUE;LOCK_MODE=0;INIT=CREATE SCHEMA IF NOT EXISTS TASKANA");
```
Comment (TitusLabs) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/RestConfiguration.java]: so, this is a testing datasource and is not correct at this place

**Comment by holgerhagen** [comment]
What are we actually aiming for? At least the taskana-rest-spring-example is a Spring Boot application, isn't it? I'm no expert in Spring but it sounds reasonable to remove the Spring Boot dependency from taskana-pring-rest.

File: rest/taskana-rest-spring/pom.xml [hunk 3]
```diff
@@ -128,52 +123,76 @@
...
 		</dependency>
-
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
-		</dependency>
-
 		<dependency>
 			<groupId>com.h2database</groupId>
```
Comment (TitusLabs) [inline rest/taskana-rest-spring/pom.xml]: should only be necessary for scope testing

Comment (mustaphazorgati) [reply rest/taskana-rest-spring/pom.xml]: done

**Pull request merged**